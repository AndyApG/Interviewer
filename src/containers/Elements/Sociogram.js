/* eslint-disable */

import React from 'react';
import PropTypes from 'prop-types';
import {
  SociogramBackground,
  NodeLayout,
  EdgeLayout,
  NodeBucket,
} from '../Elements';

const Sociogram = ({ stage, prompt }) => (
  <div className="sociogram">
    <SociogramBackground {...prompt.background} />
    {
      prompt.edges &&
      <EdgeLayout
        stage={stage}
        prompt={prompt}
      />
    }
    <NodeLayout
      stage={stage}
      prompt={prompt}
    />
    <NodeBucket
      stage={stage}
      prompt={prompt}
    />
  </div>
);

Sociogram.propTypes = {
  stage: PropTypes.object.isRequired,
  prompt: PropTypes.object.isRequired,
};

export default Sociogram;
