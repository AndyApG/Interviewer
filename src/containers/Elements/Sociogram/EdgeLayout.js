import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { colorDictionary } from 'network-canvas-ui';
import { makeDisplayEdgesForPrompt } from '../../../selectors/sociogram';

const renderEdge = (key, color, from, to) => {
  if (!from || !to) { return null; }
  return (
    <line
      key={key}
      x1={from.x}
      y1={from.y}
      x2={to.x}
      y2={to.y}
      stroke={color}
    />
  );
};

const renderEdges = (edges, color) =>
  edges.map(
    ({ key, from, to }) => renderEdge(key, color, from, to),
  );

export class EdgeLayout extends PureComponent {
  static propTypes = {
    displayEdges: PropTypes.array,
  };

  static defaultProps = {
    displayEdges: [],
  };

  render() {
    const { displayEdges } = this.props;
    const color = colorDictionary['edge-base'];

    return (
      <div className="edge-layout">
        <svg viewBox="0 0 1 1" xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="none">
          { displayEdges.map(edges => renderEdges(edges, color)) }
        </svg>
      </div>
    );
  }
}

function makeMapStateToProps() {
  const displayEdgesForPrompt = makeDisplayEdgesForPrompt();

  return function mapStateToProps(state, props) {
    return {
      displayEdges: displayEdgesForPrompt(state, props),
    };
  };
}

export default connect(makeMapStateToProps)(EdgeLayout);
