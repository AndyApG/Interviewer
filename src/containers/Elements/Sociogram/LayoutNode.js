/* eslint-disable */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import { first } from 'lodash';
import { Node } from 'network-canvas-ui';
import { selectable } from '../../../behaviours';
import { DragSource } from '../../../behaviours/DragAndDrop';

const label = node => node.nickname;

const EnhancedNode = compose(
  DragSource,
  selectable,
)(Node);

class LayoutNode extends PureComponent {
  render() {
    const {
      node,
      selected,
      allowPositioning,
      allowSelect,
      layoutVariable,
      areaWidth,
      areaHeight,
    } = this.props;

    const { x, y } = node[layoutVariable];

    const styles = {
      left: 0,
      top: 0,
      transform: `translate(calc(${x * areaWidth}px - 50%), calc(${y * areaHeight}px - 50%))`,
    };

    return (
      <div
        className="node-layout__node"
        style={styles}
      >
        <EnhancedNode
          label={label(node)}
          onSelected={this.props.onSelected}
          selected={selected}
          allowDrag={allowPositioning}
          allowSelect={allowSelect}
          meta={() => ({ ...node, itemType: 'POSITIONED_NODE' })}
          animate={false}
          {...node}
        />
      </div>
    );
  }
}

LayoutNode.propTypes = {
  onSelected: PropTypes.func.isRequired,
  layoutVariable: PropTypes.string.isRequired,
  node: PropTypes.object.isRequired,
  allowPositioning: PropTypes.bool,
  allowSelect: PropTypes.bool,
  selected: PropTypes.bool,
  areaWidth: PropTypes.number,
  areaHeight: PropTypes.number,
};

LayoutNode.defaultProps = {
  allowPositioning: false,
  allowSelect: false,
  selected: false,
  areaWidth: 0,
  areaHeight: 0,
  onSelected: (...args) => { console.log('LayoutNode.onSelected()', args) },
};

export { LayoutNode };

export default LayoutNode;
