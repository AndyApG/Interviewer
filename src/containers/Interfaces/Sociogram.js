import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withHandlers, compose, withState } from 'recompose';
import PropTypes from 'prop-types';
import withPrompt from '../../behaviours/withPrompt';
import {
  PromptObstacle,
  ButtonObstacle,
} from '../../containers/Canvas';
import { ConcentricCircles } from '../../components/Canvas';
import { actionCreators as resetActions } from '../../ducks/modules/reset';

const withConnectFrom = withState('connectFrom', 'setConnectFrom', null);

const withConnectFromHandler = withHandlers({
  handleConnectFrom: ({ setConnectFrom }) =>
    id => setConnectFrom(id),
  handleResetConnectFrom: ({ setConnectFrom }) =>
    () => setConnectFrom(null),
});

const withResetInterfaceHandler = withHandlers({
  handleResetInterface: ({
    handleResetConnectFrom,
    resetPropertyForAllNodes,
    resetEdgesOfType,
    stage,
  }) => () => {
    handleResetConnectFrom();
    stage.prompts.forEach((prompt) => {
      resetPropertyForAllNodes(prompt.layout.layoutVariable);
      if (prompt.edges) {
        resetEdgesOfType(prompt.edges.creates);
      }
    });
  },
});

/**
  * Sociogram Interface
  * @extends Component
  */
const Sociogram = ({
  promptForward,
  promptBackward,
  prompt,
  stage,
  handleResetInterface,
  connectFrom,
  handleConnectFrom,
}) => {
  const subject = prompt.subject;
  const layoutVariable = prompt.layout && prompt.layout.layoutVariable;
  const highlight = prompt.highlight && prompt.highlight.variable;
  const allowHighlight = prompt.highlight && prompt.highlight.allowHighlight;
  const createEdge = prompt.edges && prompt.edges.create;
  const allowPositioning = prompt.layout && prompt.layout.allowPositioning;
  const displayEdges = (prompt.edges && prompt.edges.display) || [];
  const backgroundImage = prompt.background && prompt.background.image;
  const concentricCircles = prompt.background && prompt.background.concentricCircles;
  const skewedTowardCenter = prompt.background && prompt.background.skewedTowardCenter;
  const sortOrder = prompt.sortOrder;
  return (
    <div className="sociogram-interface">
      <PromptObstacle
        id="PROMPTS_OBSTACLE"
        className="sociogram-interface__prompts"
        forward={promptForward}
        backward={promptBackward}
        prompts={stage.prompts}
        prompt={prompt}
        floating
        minimizable
      />
      <div className="sociogram-interface__concentric-circles">
        <ConcentricCircles
          subject={subject}
          layoutVariable={layoutVariable}
          highlight={highlight}
          allowHighlight={allowHighlight}
          createEdge={createEdge}
          allowPositioning={allowPositioning}
          displayEdges={displayEdges}
          backgroundImage={backgroundImage}
          concentricCircles={concentricCircles}
          skewedTowardCenter={skewedTowardCenter}
          sortOrder={sortOrder}
          connectFrom={connectFrom}
          updateLinkFrom={handleConnectFrom}
          key={prompt.id}
        />
      </div>
      <div style={{ position: 'absolute', right: '3rem', bottom: '3rem' }}>
        <ButtonObstacle
          id="RESET_BUTTON_OBSTACLE"
          label="RESET"
          size="small"
          onClick={handleResetInterface}
        />
      </div>
    </div>
  );
};

Sociogram.propTypes = {
  stage: PropTypes.object.isRequired,
  prompt: PropTypes.object.isRequired,
  promptForward: PropTypes.func.isRequired,
  promptBackward: PropTypes.func.isRequired,
  handleResetInterface: PropTypes.func.isRequired,
  connectFrom: PropTypes.string,
  handleConnectFrom: PropTypes.func.isRequired,
};

Sociogram.defaultProps = {
  connectFrom: null,
};

const mapDispatchToProps = dispatch => ({
  resetEdgesOfType: bindActionCreators(resetActions.resetEdgesOfType, dispatch),
  resetPropertyForAllNodes: bindActionCreators(resetActions.resetPropertyForAllNodes, dispatch),
});

export default compose(
  connect(null, mapDispatchToProps),
  withConnectFrom,
  withConnectFromHandler,
  withResetInterfaceHandler,
  withPrompt,
)(Sociogram);
