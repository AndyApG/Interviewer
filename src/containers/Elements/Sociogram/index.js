import Sociogram from './Sociogram';

export default Sociogram;
