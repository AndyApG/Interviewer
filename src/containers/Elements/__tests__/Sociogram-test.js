/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import Sociogram from '../../Elements/Sociogram';

const mockProps = {
  prompt: { sociogram: { background: {} } },
  stage: {},
};

describe('<Sociogram />', () => {
  it('renders ok', () => {
    const component = shallow(<Sociogram {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
