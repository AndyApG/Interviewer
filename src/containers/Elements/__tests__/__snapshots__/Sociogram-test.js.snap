exports[`<Sociogram /> renders ok 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sociogram">
    <SociogramBackground
        image={null}
        n={4}
        skewed={true} />
    <Connect(Component)
        attributes={null}
        edge={null}
        layout={undefined}
        position={[Function]}
        select={null} />
    <Connect(NodeBucket)
        layout={undefined}
        sort={null} />
    <div
        style={
            Object {
                "bottom": "20px",
                "position": "absolute",
                "right": 0,
              }
        }>
        <Button
            color=""
            content=""
            icon=""
            iconPosition="right"
            onClick={[Function]}
            size="">
            Reset Layout
        </Button>
    </div>
</div>,
  "nodes": Array [
    <div
      className="sociogram">
      <SociogramBackground
            image={null}
            n={4}
            skewed={true} />
      <Connect(Component)
            attributes={null}
            edge={null}
            layout={undefined}
            position={[Function]}
            select={null} />
      <Connect(NodeBucket)
            layout={undefined}
            sort={null} />
      <div
            style={
                  Object {
                        "bottom": "20px",
                        "position": "absolute",
                        "right": 0,
                      }
            }>
            <Button
                  color=""
                  content=""
                  icon=""
                  iconPosition="right"
                  onClick={[Function]}
                  size="">
                  Reset Layout
            </Button>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Sociogram
        edge={null}
        nodeAttributes={null}
        position={[Function]}
        select={null}
        sort={null}>
        Foo
</Sociogram>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": "Foo",
          "edge": null,
          "nodeAttributes": null,
          "position": [Function],
          "select": null,
          "sort": null,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="sociogram">
          <SociogramBackground
                    image={null}
                    n={4}
                    skewed={true} />
          <Connect(Component)
                    attributes={null}
                    edge={null}
                    layout={undefined}
                    position={[Function]}
                    select={null} />
          <Connect(NodeBucket)
                    layout={undefined}
                    sort={null} />
          <div
                    style={
                              Object {
                                        "bottom": "20px",
                                        "position": "absolute",
                                        "right": 0,
                                      }
                    }>
                    <Button
                              color=""
                              content=""
                              icon=""
                              iconPosition="right"
                              onClick={[Function]}
                              size="">
                              Reset Layout
                    </Button>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="sociogram">
          <SociogramBackground
                    image={null}
                    n={4}
                    skewed={true} />
          <Connect(Component)
                    attributes={null}
                    edge={null}
                    layout={undefined}
                    position={[Function]}
                    select={null} />
          <Connect(NodeBucket)
                    layout={undefined}
                    sort={null} />
          <div
                    style={
                              Object {
                                        "bottom": "20px",
                                        "position": "absolute",
                                        "right": 0,
                                      }
                    }>
                    <Button
                              color=""
                              content=""
                              icon=""
                              iconPosition="right"
                              onClick={[Function]}
                              size="">
                              Reset Layout
                    </Button>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Sociogram
    edge={null}
    nodeAttributes={null}
    position={[Function]}
    select={null}
    sort={null}>
    Foo
</Sociogram>,
}
`;
